smt.go

```diff
@@ -142,20 +142,31 @@ func (tree *BASSparseMerkleTree) initFromStorage() error {
 }
 
 func (tree *BASSparseMerkleTree) extendNode(node *TreeNode, nibble, path uint64, depth uint8) error {
-	if node.Children[nibble] == nil {
-		node.Children[nibble] = NewTreeNode(depth, path, tree.nilHashes, tree.hasher)
+	err := tree.constructNode(node, nibble, path, depth)
+	if err != nil {
+		return err
 	}
-	if depth < tree.maxDepth && // no need to extend leaf nodes
-		!node.Children[nibble].Extended() {
-		err := tree.constructNode(node, nibble, path, depth)
-		if err != nil {
-			return err
-		}
+
+	if depth < tree.maxDepth && node.Children[nibble^1] != nil {
```

Comment:
Hard to understand.

Reply:
After discussion, we simplified the logic and removed this function